app/controllers/application_controller.rb

```diff
@@ -1,7 +1,41 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  def set_locale
+    if user_signed_in? && !current_user.language.blank?
+      I18n.locale = current_user.language
+    else
+      I18n.locale =
+        if params[:locale].present? && available_locale?(params[:locale])
+          params[:locale]
+        else
+          extract_locale_from_accept_language_header
+        end
+      if user_signed_in? && current_user.language.blank?
```

Comment:
Might be nice to avoid checking `user_signed_in?` again by adjusting branching logic.